res_syntax/src/reactjs_jsx_ppx.ml

```diff
@@ -127,8 +127,8 @@ let getMapper ~config =
           | Pstr_attribute attr -> processConfigAttribute attr config
           | _ -> ());
           let item = default_mapper.structure_item mapper item in
-          if config.version = 3 then transformStructureItem3 mapper item
-          else if config.version = 4 then transformStructureItem4 mapper item
```

Comment:
This was applying the JSX transformation recursively again, to some parts of the structure item.